### PR TITLE
Update reverse translation for coredns

### PR DESCRIPTION
**Pull request by richardwxn**

### CODE REVIEW

File: pkg/translate/translate_value_test.go [hunk 1]
```diff
@@ -326,6 +337,11 @@ telemetry:
    telemetry:
      enabled: false
  enabled: false
+coreDNS:
+ components:
+   coreDNS:
```
Comment (howardjohn) [inline pkg/translate/translate_value_test.go]: This feels weird

Comment (richardwxn) [reply pkg/translate/translate_value_test.go]: can you clarify?

Comment (richardwxn) [reply pkg/translate/translate_value_test.go]: in this case, the source values.yaml didn't enable coreDNS so it is expected.

Comment (howardjohn) [reply pkg/translate/translate_value_test.go]: Not necessarily and issue with this pr, but it makes no sense to do `--set coreDNS.enabled=true --set coreDNS.components.coreDNS.enabled=true`. Its so confusing for users that we have the same thing enabled in two places.

I don't know how our users will ever figure out the magic settings to get it to work

Comment (ostromart) [reply pkg/translate/translate_value_test.go]: That should not be necessary. Just --set coreDNS.enabled=true, the reason both were needed before was due to a bug.

Comment (richardwxn) [reply pkg/translate/translate_value_test.go]: yes the component enablement would inherit the feature one, unless you need to override

**Comment by richardwxn** [comment]
@howardjohn PTAL

**Review by ostromart — APPROVED** [review]

**Pull request merged**